### PR TITLE
Feature/apl 604 height monitor server

### PR DESCRIPTION
**Pull request by AndrewBoyarsky**

### CODE REVIEW

File: apl-tools/src/main/java/com/apollocurrency/aplwallet/apl/tools/impl/heightmon/HeightMonitor.java [hunk 1]
```diff
@@ -85,97 +52,12 @@ public void start() {
 
     public void stop() {
         try {
+            container.shutdown();
             executor.shutdown();
```
Comment (yuriy-larin-firstbridge) [inline apl-tools/src/main/java/com/apollocurrency/aplwallet/apl/tools/impl/heightmon/HeightMonitor.java]: Should we shutdown executor at first and container at second ??

File: apl-tools/src/main/java/com/apollocurrency/aplwallet/apl/tools/impl/heightmon/HeightMonitorServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,270 @@
...
+    public NetworkStats getLastStats() {
+        return lastStats.get();
+    }
+
+    @Override
+    public NetworkStats updateStats() {
+        log.info("===========================================");
+        Map<String, List<Block>> peerBlocks = getPeersBlocks();
+        log.info(String.format("%5.5s %5.5s %-16.16s %-16.16s %9.9s %7.7s %7.7s", "diff1", "diff2", "peer1", "peer2", "milestone", "height1", "height2"));
+        int currentMaxBlocksDiff = -1;
```
Comment (yuriy-larin-firstbridge) [inline apl-tools/src/main/java/com/apollocurrency/aplwallet/apl/tools/impl/heightmon/HeightMonitorServiceImpl.java]: Could you pls add 'VERSION' column here (and into JSON)? Sometime it's very useful.

File: apl-tools/src/main/java/com/apollocurrency/aplwallet/apl/tools/impl/heightmon/HeightMonitorServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,270 @@
...
+                Block lastMutualBlock = findLastMutualBlock(blocksToCompare, targetBlocks);
+                if (!blocksToCompare.isEmpty()) {
+                    int lastHeight = targetBlocks.get(0).getHeight();
+                    int blocksDiff1 = getBlockDiff(lastMutualBlock, lastHeight);
+                    int blocksDiff2 = getBlockDiff(lastMutualBlock, blocksToCompare.get(0).getHeight());
+                    int milestoneHeight = getMilestoneHeight(lastMutualBlock);
+                    currentMaxBlocksDiff = Math.max(blocksDiff1, currentMaxBlocksDiff);
+                    log.info(String.format("%5d %5d %-16.16s %-16.16s %9d %7d %7d", blocksDiff1, blocksDiff2, host1, host2, milestoneHeight, lastHeight, blocksToCompare.get(0).getHeight()));
+                    networkStats.getPeerDiffStats().add(new PeerDiffStat(blocksDiff1, blocksDiff2, host1, host2, lastHeight, milestoneHeight, blocksToCompare.get(0).getHeight()));
+                }
```
Comment (yuriy-larin-firstbridge) [inline apl-tools/src/main/java/com/apollocurrency/aplwallet/apl/tools/impl/heightmon/HeightMonitorServiceImpl.java]: what about 'VERSION' ??

File: apl-tools/src/main/java/com/apollocurrency/aplwallet/apl/tools/impl/heightmon/HeightMonitor.java (hunk 1 above)
Comment (AndrewBoyarsky) [reply apl-tools/src/main/java/com/apollocurrency/aplwallet/apl/tools/impl/heightmon/HeightMonitor.java]: fixed

File: apl-tools/src/main/java/com/apollocurrency/aplwallet/apl/tools/impl/heightmon/HeightMonitorServiceImpl.java (hunk 2 above)
Comment (AndrewBoyarsky) [reply apl-tools/src/main/java/com/apollocurrency/aplwallet/apl/tools/impl/heightmon/HeightMonitorServiceImpl.java]: Added

File: apl-tools/src/main/java/com/apollocurrency/aplwallet/apl/tools/impl/heightmon/HeightMonitorServiceImpl.java (hunk 3 above)
Comment (AndrewBoyarsky) [reply apl-tools/src/main/java/com/apollocurrency/aplwallet/apl/tools/impl/heightmon/HeightMonitorServiceImpl.java]: Added

**Pull request merged**